File: crates/normalize-facts/tests/fixtures/java/add_numbers/project/Calculator.java
import java.util.ArrayList;
import java.util.List;

public class Calculator {
    private List<Integer> history;

    public Calculator() {
        this.history = new ArrayList<>();
    }

    public int add(int a, int b) {
        int result = a + b;
        history.add(result);
        return result;
    }

    public int multiply(int a, int b) {
        int result = a * b;
        history.add(result);
        return result;
    }

    public List<Integer> getHistory() {
        return history;
    }

    public static void main(String[] args) {
        Calculator calc = new Calculator();
        System.out.println(calc.add(2, 3));
        System.out.println(calc.multiply(4, 5));
    }
}

File: crates/normalize-facts/tests/fixtures/php/add_numbers/project/Calculator.php
<?php

namespace App\Math;

use App\Logging\Logger;
use InvalidArgumentException;

interface MathOperation {
    public function execute(int $a, int $b): int;
}

function add(int $a, int $b): int {
    return $a + $b;
}

class Calculator implements MathOperation {
    private array $history;
    private Logger $logger;

    public function __construct(Logger $logger) {
        $this->history = [];
        $this->logger = $logger;
    }

    public function execute(int $a, int $b): int {
        $result = add($a, $b);
        $this->history[] = $result;
        $this->logger->log("Result: $result");
        return $result;
    }

    public function multiply(int $a, int $b): int {
        $result = $a * $b;
        $this->history[] = $result;
        return $result;
    }

    public function getHistory(): array {
        return $this->history;
    }
}
